File: ui/article-renderer/src/main/java/org/cru/godtools/article/databinding/adapter/TextViewAdapters.java
package org.cru.godtools.article.databinding.adapter;

import android.widget.TextView;

import org.cru.godtools.base.tool.model.view.TextViewUtils;
import org.cru.godtools.xml.model.Text;

import androidx.annotation.ColorInt;
import androidx.annotation.DimenRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.BindingAdapter;

public final class TextViewAdapters {
    @BindingAdapter(value = {"textNode", "android:textSize"})
    public static void setText(@NonNull final TextView view, @Nullable final Text text,
                               @Nullable @DimenRes final Integer textSize) {
        TextViewUtils.bind(text, view, textSize, null);
    }

    @BindingAdapter(value = {"textNode", "android:textSize", "defaultTextColor"})
    public static void setText(@NonNull final TextView view, @Nullable final Text text,
                               @Nullable @DimenRes final Integer textSize,
                               @Nullable @ColorInt final Integer defaultTextColor) {
        TextViewUtils.bind(text, view, textSize, defaultTextColor);
    }
}

File: ui/article-renderer/src/main/java/org/cru/godtools/article/databinding/adapter/TextViewAdapters.kt
package org.cru.godtools.article.databinding.adapter

import android.widget.TextView
import androidx.annotation.ColorInt
import androidx.annotation.DimenRes
import androidx.databinding.BindingAdapter
import org.cru.godtools.base.tool.model.view.TextViewUtils
import org.cru.godtools.xml.model.Text

@BindingAdapter(value = ["textNode", "android:textSize"])
internal fun TextView.setTextNode(text: Text?, @DimenRes textSize: Int?) =
    TextViewUtils.bind(text, this, textSize, null)

@BindingAdapter(value = ["textNode", "android:textSize", "defaultTextColor"])
internal fun TextView.setTextNode(text: Text?, @DimenRes textSize: Int?, @ColorInt defaultTextColor: Int?) =
    TextViewUtils.bind(text, this, textSize, defaultTextColor)
